src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContextHelpers.cs

```diff
@@ -79,7 +82,7 @@ internal static void VerifyDiagnosticLocationsInCompilation(Diagnostic diagnosti
             {
                 foreach (var location in diagnostic.AdditionalLocations)
                 {
-                    VerifyDiagnosticLocationInCompilation(diagnostic.Id, diagnostic.Location, compilation);
+                    VerifyDiagnosticLocationInCompilation(diagnostic.Id, location, compilation);
```

Comment:
📝 This was a bug fix. Do we have new tests covering it?

Reply:
Ah thanks! I meant to pull this out into a separate PR with just this change + regression test. I will move this out.

Reply:
Filed https://github.com/dotnet/roslyn/issues/46728